packages/microcosm/src/agent.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @fileoverview An agent is a general abstraction for intercepting actions.
+ * It is the basis of Domain and Effect
+ * @flow
+ */
+
+import { type Microcosm } from '../src/microcosm'
+import { Subject } from './subject'
+import { merge } from './data'
+
+export class Agent extends Subject {
+  repo: Microcosm
+  options: Object
+
+  static defaults: ?Object
+
+  constructor(repo: Microcosm, options?: Object) {
```

Comment:
Thinking on this a bit more, I'm also curious about the idea of an `AgentPool` class that could distribute state management across a group of presenters by sharing agents with the same configuration.

Reply:
You'd do that by deriving a hashcode from the repo's identity and values passed via options